WordPress/pattern-directory
# Theme: Hook up pattern favorite functionality

Fixes #52. This PR is based on #93, which sets up the API, and this PR sets up the data store and hooks up the actions to the `FavoriteButton`.

The `FavoriteButton` now has two props, `patternId` (required) and `showLabel` (defaults to true). This component handles fetching favorites, checking if this pattern (`patternId`) is favorited, and triggering the add/remove from favorite actions. `showLabel` is used to show/hide the "Add to favorites" text when we want to show a smaller button. The label is still available to screen reader users.

### Screenshots

<img width="330" alt="Screen Shot 2021-05-05 at 5 58 43 PM" src="https://user-images.githubusercontent.com/541093/117214999-98e45800-adcb-11eb-9e28-f32eb070326f.png">
<img width="644" alt="Screen Shot 2021-05-05 at 5 58 54 PM" src="https://user-images.githubusercontent.com/541093/117215000-98e45800-adcb-11eb-8699-06a9829aa6c6.png">
<img width="601" alt="Screen Shot 2021-05-05 at 5 59 01 PM" src="https://user-images.githubusercontent.com/541093/117215001-997cee80-adcb-11eb-82cb-2fab3016941c.png">

### How to test the changes in this Pull Request:

1. Visit the patterns list view, logged in
2. Try favoriting some patterns, they should turn red
3. Click into any favorited pattern, the button in the single pattern view should also be red
4. Unfavoriting patterns should also work, and persist around the site

If you're logged out, the buttons will still be there, but they're disabled (since favorites need a user). We can also just remove them or add a tooltip - wasn't sure which would be better.
